submissions/kotlyar-andrey/a-tiny-JS-world/index.js

```diff
@@ -5,72 +5,132 @@
    Web app: https://kotlyar-andrey.github.io/a-tiny-JS-world/
    */
 
-class Creature {
-  constructor(species, name, gender, legs, hands, phrase, linkTo = null) {
+/**
+ * Общий класс для всех жителей
+ */
+ class Creature {
+  constructor(species, name, gender, phrase) {
     this._species = species;
     this._name = name;
     this._gender = gender;
-    this._legs = legs;
-    this._hands = hands;
     this._phrase = phrase;
     this._friends = [];
-    this._link = linkTo;
-    this._properties = [
-      "species",
-      "_name",
-      "_gender",
-      "_legs",
-      "_hands",
-      "saying",
-      "friends",
-    ];
-  }
-
-  get species() {
-    return this._link
-      ? `${this._species}-${this._link._species}`
-      : this._species;
+    this._properties = ["_species", "_name", "_gender"];
   }
 
-  get friends() {
+  getFriends() {
     return this._friends && this._friends.length > 0
       ? this._friends.map(friend => friend._name).join(", ")
       : "no friends";
   }
 
-  set friends(newFriends) {
+  setFriends(newFriends) {
     this._friends = [...newFriends];
   }
 
-  get saying() {
-    return this._link ? this._link._phrase : this._phrase;
+  getSaying() {
+    return this._phrase;
+  }
+
+  setSaying(newPhrase) {
+    this._phrase = newPhrase;
+  }
+
+  addProperties(...newProperties) {
+    this._properties = [...this._properties, ...newProperties];
+  }
+
+  getPropertiesData() {
+    return this._properties.map(prop => this[prop]).join("; ");
+  }
+
+  getAnotherData() {
+    const friends = this.getFriends();
+    const saying = this.getSaying();
+    return `; ${friends}; ${saying}`;
+  }
+
+  getInfo() {
+    return this.getPropertiesData() + this.getAnotherData();
+  }
+}
+
+/**
+ * Общий класс для всех животных с ногами/лапами
+ */
+class Animal extends Creature {
+  constructor(species, name, gender, phrase, legs) {
+    super(species, name, gender, phrase);
+    this._legs = legs;
+    this.addProperties("_legs");
+  }
```

Comment:
If some class (e.g. `Creature`) is extended by one and only one class (e.g. `Animal`) this means they can be effectively merged. Yes, we may have legless creatures in future, but that's exactly the point when we would split a base class and/or add yet another base class on the top of hierarchy tree.